pkg/remote/trans/default_server_handler.go

```diff
@@ -158,10 +159,9 @@ func (t *svrTransHandler) OnRead(ctx context.Context, conn net.Conn) (err error)
 		if rpcinfo.PoolEnabled() {
 			t.opt.InitOrResetRPCInfoFunc(ri, conn.RemoteAddr())
 		}
-		if wrapErr != nil {
-			err = wrapErr
-		}
 		if err != nil && !closeConnOutsideIfErr {
+			// when error is not nil, outside will close conn,
+			// set err not nil just indicate that this kind of error don't need close conn
```

Comment:
set `err` to nil to indicate that this kind of error does not require closing the connection

Reply:
done